core/mount/mount_linux_test.go

```diff
@@ -198,6 +206,120 @@ func TestUnmountRecursive(t *testing.T) {
 	}
 }
 
+func TestDoPrepareIDMappedOverlay(t *testing.T) {
+	testutil.RequiresRoot(t)
+
+	k512 := kernel.KernelVersion{Kernel: 5, Major: 12}
+	ok, err := kernel.GreaterEqualThan(k512)
+	require.NoError(t, err)
+	if !ok {
+		t.Skip("GetUsernsFD requires kernel >= 5.12")
```

Comment:
The propper way to test for support should be something like this: https://github.com/containerd/containerd/blob/main/integration/pod_userns_linux_test.go#L51 (just c&p seems fine).

I think I'd use that, because idmap support depends on the fs having support. The kernel 5.12 added the first supported fs, but overlay was added in 5.19, we added tmpfs in 6.3, etc.

It is quite common to use tmpdirs in /tmp for tests, and if we upgrade the CI to a distro that switched to tmpfs in /tmp and the kernel it is using doesn't support idmap mounts for tmpfs, then the test will fail when we upgrade the CI. It will not be super simple to debug.

If we just test that the paths we will use support it, then those issues can't happen.

Reply:
Happy to change it. Worth noting that the `mount` package is doing other unit tests of idmapping already though using a kernel version check, here: https://github.com/containerd/containerd/blob/2b89ebb3f6a08e72c8c7ef561770feaffb528577/core/mount/mount_idmapped_linux_test.go#L77

Reply:
I added the linked code.